### DISCUSSION TITLE
how to use AI assistant

### DISCUSSION THREAD

**Discussion by meisongdai** [chhoumann/quickadd#485] [Q&A] [answered]
I didn't find any information in the documentation, so I came to ask for help

**Comment by chhoumann** [comment, marked as answer]
https://bagerbach.com/blog/obsidian-ai

Documentation is there for now. LMK if you have further questions!

**Reply by beesinblenders** [reply]
Hey, something I can't seem to find: how can I pass the text of the current note as context to the LLM? (I saw that I could use {{selected}} but that's not exactly what I need)

**Reply by ces3001** [reply]
Used a prompt template, and in that template do something like this:

The following is a description of one or more dreams. Please interpret the dream(s), giving at least 3 or 4 different perspectives on its meaning from Western, Jungian, Buddhist, Sangoma, and Shamanic dream interpretation traditions. Names I use are: (x is my wife, y my son, z my brother, etc.) At the end, please suggest a short contemplation to face the spiritual or psychological material in the dream. Please use fourth-level headings (`####`) for each part. Here is the dream(s):
```js quickadd
const noteFile = app.workspace.getActiveFile(); // Currently Open Note
if(!noteFile.name) return; // Nothing Open
let text = await app.vault.cachedRead(noteFile); // Read the contents
//const regex = /type:\s*dream/i; // Check if it's a Dream. 'i' flag for case-insensitive matching
//if (!regex.test(text)) return "This is not a dream."
return text; // return contents 
```

**Comment by chhoumann** [comment]
Late update for anyone landing here: the AI Assistant has proper documentation now at https://quickadd.obsidian.guide/docs/AIAssistant - that supersedes the blog post I linked back then. And for the follow-up in the replies about passing the current note as context: reading the active note inside a ```js quickadd``` block in the prompt template, as @ces3001 showed, is still the way to do it. Marking as answered and closing.
